targets/TARGET_NUVOTON/TARGET_M451/can_api.c

```diff
@@ -0,0 +1,306 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+ #include "can_api.h"
+ #include "m451_gpio.h"
+ #include "m451_can.h"
+ 
+ #if DEVICE_CAN
+ #include <string.h>
+ #include "cmsis.h"
+ #include "pinmap.h"
+ #include "PeripheralPins.h"
+ #include "nu_modutil.h"
+ #include "nu_miscutil.h"
+ #include "nu_bitutil.h"
+ #include "critical.h"
+ 
+ #define NU_CAN_DEBUG    0
+ #define CAN_NUM		 1
+ 
+ static uint32_t can_irq_ids[CAN_NUM] = {0};
+ static can_irq_handler can0_irq_handler;
+
+ 
+ static const struct nu_modinit_s can_modinit_tab[] = {
+    {CAN_0, CAN0_MODULE, 0, 0, CAN0_RST, CAN0_IRQn, NULL},
+    
+    {NC, 0, 0, 0, 0, (IRQn_Type) 0, NULL}
+};
+
+ 
+ void can_init(can_t *obj, PinName rd, PinName td)
+ {
+	uint32_t can_td = (CANName)pinmap_peripheral(td, PinMap_CAN_TD);
+    uint32_t can_rd = (CANName)pinmap_peripheral(rd, PinMap_CAN_RD);
+    obj->can = (CANName)pinmap_merge(can_td, can_rd);
+    MBED_ASSERT((int)obj->can != NC);
+    
+    const struct nu_modinit_s *modinit = get_modinit(obj->can, can_modinit_tab);
+    MBED_ASSERT(modinit != NULL);
+    MBED_ASSERT(modinit->modname == obj->can);
+    
+    // Reset this module
+    SYS_ResetModule(modinit->rsetidx);
+    
+    // Enable IP clock
+    CLK_EnableModuleClock(modinit->clkidx);
+	 
+	obj->index = 0;
+	
+    pinmap_pinout(td, PinMap_CAN_TD);
+    pinmap_pinout(rd, PinMap_CAN_RD);
+    
+	/* For M453 mbed Board Transmitter Setting (RS Pin) */
+    GPIO_SetMode(PA, BIT0| BIT1, GPIO_MODE_OUTPUT);    
+    PA0 = 0x00;
+    PA1 = 0x00;   
+
+	CAN_Open((CAN_T *)obj->can, 500000, CAN_NORMAL_MODE);
+    
+    can_filter(obj, 0, 0, CANStandard, 0);
+ }
+ 
+ 
+void can_free(can_t *obj)
+{
+
+	const struct nu_modinit_s *modinit = get_modinit(obj->can, can_modinit_tab);
+	
+	MBED_ASSERT(modinit != NULL);
+    MBED_ASSERT(modinit->modname == obj->can);
+    
+    // Reset this module
+    SYS_ResetModule(modinit->rsetidx);
+	
+	CLK_DisableModuleClock(modinit->clkidx);
+}
+
+int can_frequency(can_t *obj, int hz)
+{
+	CAN_SetBaudRate((CAN_T *)obj->can, hz);
+	
+	return CAN_GetCANBitRate((CAN_T *)obj->can);
+}
+
+static void can_irq(CANName name, int id) 
+{
+	
+	CAN_T *can = (CAN_T *)NU_MODBASE(name);
+	uint32_t u8IIDRstatus;
+
+    u8IIDRstatus = can->IIDR;
+
+    if(u8IIDRstatus == 0x00008000) {      /* Check Status Interrupt Flag (Error status Int and Status change Int) */
+        /**************************/
+        /* Status Change interrupt*/
+        /**************************/
+        if(can->STATUS & CAN_STATUS_RXOK_Msk) {
+            can->STATUS &= ~CAN_STATUS_RXOK_Msk;   /* Clear Rx Ok status*/
+			can0_irq_handler(can_irq_ids[id], IRQ_RX);
+        }
+
+        if(can->STATUS & CAN_STATUS_TXOK_Msk) {
+            can->STATUS &= ~CAN_STATUS_TXOK_Msk;    /* Clear Tx Ok status*/
+			can0_irq_handler(can_irq_ids[id], IRQ_TX);
```

Comment:
misaligned, some lines are . ``can_init``, ``can_free`` for instance contain lot of misaligned lines